.github/workflows/workflow.yml

```diff
@@ -2,149 +2,102 @@ name: workflow
 
 on: [push, pull_request]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   lock:
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
     runs-on: "ubuntu-latest"
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
       - run: curl -sSL https://install.python-poetry.org | python3 -
       - run: poetry lock
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         with:
           path: poetry.lock
           key: ${{ github.sha }}-${{ matrix.python-version }}
-  pytest-dynalite:
-    needs: lock
-    strategy:
-      matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
-    runs-on: "ubuntu-latest"
-    services:
-      dynalite:
-        image: ghcr.io/hennge/dynalite:latest
-        ports:
-          - 8000:8000
-    env:
-      DYNAMODB_URL: http://localhost:8000
-      AWS_ACCESS_KEY_ID: dummy
-      AWS_SECRET_ACCESS_KEY: dummy
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.python-version }}
-      - uses: actions/cache@v2
-        with:
-          path: poetry.lock
-          key: ${{ github.sha }}-${{ matrix.python-version }}
-      - run: curl -sSL https://install.python-poetry.org | python3 -
-      - run: poetry install --extras aiohttp --extras httpx
-      - run: poetry run pytest --verbose
-  pytest-dynamodb-local:
+
+  pytest:
     needs: lock
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+        implementation: [{flavor: "other", name: "dynalite"}, {flavor: "other", name: "dynamodb-local"}, {flavor: "other", name: "localstack"}, {flavor: "scylla", name: "scylla"}]
     runs-on: "ubuntu-latest"
-    services:
-      dynamodb:
-        image: amazon/dynamodb-local:latest
-        ports:
-          - 8000:8000
     env:
       DYNAMODB_URL: http://localhost:8000
+      DYNAMODB_FLAVOR: ${{ matrix.implementation.flavor }}
+      DYNAMODB_NAME: ${{ matrix.implementation.name }}
       AWS_ACCESS_KEY_ID: dummy
       AWS_SECRET_ACCESS_KEY: dummy
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - run: docker compose -f .github/docker-compose/docker-compose.${{ matrix.implementation.name }}.yml up -d
```

Comment:
Since you already set an env for it, you can just use that instead:
```yml
- run: docker compose -f .github/docker-compose/docker-compose.${DYNAMODB_NAME}.yml up -d
```

Reply:
tbh i prefer the current way it's written.

Reply:
But currently we need to put `${{ matrix.implementation.name }}` two times right?

Reply:
yea but it makes it clear to me where this value is coming from